packages/block-library/src/search/view.js

```diff
@@ -66,6 +81,41 @@ const { actions } = store(
 					actions.closeSearchInput();
 				}
 			},
+			*updateSearch() {
+				const { ref } = getElement();
+				const { value } = ref;
+
+				// Don't navigate if the search didn't really change.
+				if ( value === state.search ) {
+					return;
+				}
+
+				state.search = value;
+
+				// Debounce the search by 300ms to prevent multiple navigations.
+				// We can do this by yielding a promise that resolves after 300ms and
+				// then bailing out if the search has changed.
+				yield new Promise( ( resolve ) => setTimeout( resolve, 300 ) );
+				if ( value !== state.search ) {
+					return;
+				}
+
+				const url = new URL( window.location );
+				if ( ! isEmpty( value ) ) {
+					state.search = value;
+					url.searchParams.set( 'instant-search', value );
+				} else {
+					url.searchParams.delete( 'instant-search' );
+				}
+
+				const { actions: routerActions } = yield import(
+					'@wordpress/interactivity-router'
+				);
+
+				routerActions.navigate(
+					`${ window.location.pathname }${ url.search }`
+				);
```

Comment:
I don't think we should need to concatenate the URL string like this, but just use the url that's been constructed:

```suggestion
				routerActions.navigate( url.href );
```

Reply:
done in 818bc96a30ab3d944d0c6514032c6df9dbcaed6a